client/state/initial-state.js

```diff
@@ -94,8 +104,10 @@ export function persistOnChange( reduxStore, serializeState = serialize ) {
 }
 
 export default function createReduxStoreFromPersistedInitialState( reduxStoreReady ) {
-	if ( config.isEnabled( 'persist-redux' ) && ! isSupportUserSession() ) {
-		localforage.getItem( 'redux-state' )
+	if ( config.isEnabled( 'persist-redux' ) && isLoggedIn() && ! isSupportUserSession() ) {
+		// clear storage any storage that used the old-style key
+		localforage.removeItem( 'redux-state' );
```

Comment:
@gwwar Do you think we need this line to remove old-style storage? Better to do this somewhere else?